lib/repositories/pages_repository.rb

```diff
@@ -31,11 +40,19 @@ def update(page)
   end
 
   def delete(page_id)
-    @database[:pages].where(id: page_id).delete
+    @database.transaction(isolation: :serializable) do
+      pages = @database[:pages]
+
+      # get the next value of our page and update any pages which pointed to us
+      # to that instead
```

Comment:
Think this also needs to account for the case where the first page in a form is deleted - so the value that needs updating is the parent form's `start_page`

Reply:
Looks like you're right about `start_page` just returning the lowest id so I'll resolve this and approve. At some point (probably when we add page reordering?) we'll need to change it to explicitly update `start_page` when the first page is created / deleted but don't need to worry about that yet